saleor/checkout/base_calculations.py

```diff
@@ -256,32 +294,32 @@ def apply_checkout_discount_on_checkout_line(
         or voucher.apply_once_per_order
         or voucher.type in [VoucherType.SHIPPING, VoucherType.SPECIFIC_PRODUCT]
     ):
-        return line_unit_price
+        return line_price
 
     line_quantity = checkout_line_info.line.quantity
     total_discount_amount = checkout_info.checkout.discount_amount
-    line_total_price = line_unit_price * line_quantity
+    line_total_price = line_price * line_quantity if is_unit_price else line_price
```

Comment:
Question: Wouldn't be better to always pass `line_total_price` to `apply_checkout_discount_on_checkout_line` instead of using `is_unit_price` flag? It looks like the perfect place for making the mistake 😄  In places where we call this function, and where it uses a unit price, we could simply call unit_price* quantity, right?

Reply:
You're right! I'll change that!